detectable/src/main/java/com/synopsys/integration/detectable/detectables/carthage/CarthageDetectable.java

```diff
@@ -35,20 +34,25 @@ public CarthageDetectable(DetectableEnvironment environment, FileFinder fileFind
     @Override
     public DetectableResult applicable() {
         Requirements requirements = new Requirements(fileFinder, environment);
-        requirements.eitherFile(CARTFILE_FILENAME, CARTFILE_RESOLVED_FILENAME, foundCartfile -> cartfile = foundCartfile, foundCartfileResolved -> cartfileResolved = foundCartfileResolved);
+        requirements.eitherFile(
+            CARTFILE_FILENAME,
+            CARTFILE_RESOLVED_FILENAME,
+            foundCartfile -> {}, // Used only to identify Carthage projects
+            foundCartfileResolved -> cartfileResolved = foundCartfileResolved
+        );
         return requirements.result();
     }
 
     @Override
     public DetectableResult extractable() throws DetectableException {
-        if (cartfileResolved == null && cartfile != null) {
-            return new CartfileResolvedNotFoundDetectableResult(environment.getDirectory().getAbsolutePath());
-        }
-        return new PassedDetectableResult();
+        Requirements requirements = new Requirements(fileFinder, environment);
+        requirements.optionalFile(CARTFILE_FILENAME);
+        requirements.file(CARTFILE_RESOLVED_FILENAME, new CartfileResolvedNotFoundDetectableResult(environment.getDirectory().getAbsolutePath()));
```

Comment:
Would it make sense to not have the optional file here? We just require the resolved actually exists right?

Reply:
Yeah I wasn't sure about that. It doesn't really matter. Are relevant files for diagnostics taken from the isApplicable or Extraction? Is this file even relevant?
I'm fine with removing it from here.